Fix editor crash

reaperrr opened the pull request:
Fixes #14995.

pchote commented:
An alternative approach that would be worth considering is to split out a ^BasePlayer the same way we do for the world, and avoid initializing these traits completely in the editor.

abcdefg30 commented:
> An alternative approach that would be worth considering is to split out a ^BasePlayer the same way we do for the world, and avoid initializing these traits at all in the editor.

Imho that would be the preferred approach, as just using default values when certain traits are null feels like a hack/workaround. I agree that we don't need to initialize traits like `ConquestVictoryConditions` in the map editor.
The second commit should be fine, however.

reaperrr committed:
Add EditorPlayer to all mods

To prevent the editor from loading unnecessary or even incompatible
player traits.

reaperrr commented:
Updated.

abcdefg30 commented on OpenRA.Mods.Common/UpdateRules/Rules/AddEditorPlayer.cs:
```diff
@@ -0,0 +1,34 @@
+		{
+			get
+			{
+				return "Map editor now requires an EditorPlayer to avoid loading all kinds of unnecessary player traits.\n";
```
No `\n` at the end of the message, please.

abcdefg30 commented on OpenRA.Mods.Common/UpdateRules/Rules/AddEditorPlayer.cs:
```diff
@@ -0,0 +1,34 @@
+
+		public override IEnumerable<string> AfterUpdate(ModData modData)
+		{
+			yield return "The map editor now requires an EditorPlayer actor. " +
```
You need to add a `\n` instead of the white space at the end here. Using `+` doesn't automatically add a line break. (In case you wanted a line break here, as this message is quite long?)

pchote replied:
Please define and toggle a bool so that this message is only reported the first time around (which is in the mod rules), instead of being repeated for all maps.

abcdefg30 reviewed (dismissed):
Looks good to me. 👍 

Do we want to merge this before we can turn off the update warnings on maps?

reaperrr commented:
Updated.

pchote commented on OpenRA.Mods.Common/Traits/Player/ConquestVictoryConditions.cs:
```diff
@@ -32,6 +32,7 @@ public class ConquestVictoryConditionsInfo : ITraitInfo, Requires<MissionObjecti
 	public class ConquestVictoryConditions : ITick, INotifyObjectivesUpdated
 	{
 		readonly ConquestVictoryConditionsInfo info;
+		readonly Player player;
```
Do you expect this to ever be anything except `self.Owner`?

reaperrr commented:
Updated, dropped the ConquestVictoryConditions changes as discussed on IRC.

GraionDilach reviewed (approved):
:+1:

reaperrr commented:
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/39c12a09e41350f839ca4916ff307f915bec450f...3cdaa3b8b8cda92242122bf427f1e2e2c41fe4f1
